nilearn/maskers/surface_maps_masker.py

```diff
@@ -0,0 +1,426 @@
+        standardize=False,
+        standardize_confounds=True,
+        detrend=False,
+        high_variance_confounds=False,
+        low_pass=None,
+        high_pass=None,
+        t_r=None,
+        memory=None,
+        memory_level=1,
+        verbose=0,
+        reports=True,
+        cmap="inferno",
+        clean_args=None,
+    ):
+        self.maps_img = maps_img
+        self.mask_img = mask_img
+        self.allow_overlap = allow_overlap
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.standardize_confounds = standardize_confounds
+        self.high_variance_confounds = high_variance_confounds
+        self.detrend = detrend
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.memory = memory
+        self.memory_level = memory_level
+        self.verbose = verbose
+        self.reports = reports
+        self.cmap = cmap
+        self.clean_args = clean_args
+        self._shelving = False
+        # content to inject in the HTML template
+        self._report_content = {
+            "description": (
+                "This report shows the input surface image "
+                "(if provided via img) overlaid with the regions provided via "
+                "maps_img."
+            ),
+            "n_vertices": {},
+            "number_of_regions": 0,
+            "summary": {},
+        }
+
+    def fit(self, img=None, y=None):
+        """Prepare signal extraction from regions.
+
+        Parameters
+        ----------
+        img : :obj:`~nilearn.surface.SurfaceImage` object or None, default=None
+            This parameter is currently unused.
+
+        y : None
+            This parameter is unused.
+            It is solely included for scikit-learn compatibility.
+
+        Returns
+        -------
+        SurfaceMapsMasker object
+        """
+        del img, y
+
+        logger.log(
+            msg=f"loading regions from {self.maps_img.__repr__()}",
+            verbose=self.verbose,
+        )
+        # check maps_img data is 2D
+        check_surface_data_ndims(self.maps_img, 2, "maps_img")
+        self.maps_img_ = self.maps_img.copy()
+
+        self.n_elements_ = self.maps_img.shape[1]
+
+        if self.mask_img is not None:
+            logger.log(
+                msg=f"loading regions from {self.mask_img.__repr__()}",
+                verbose=self.verbose,
+            )
+            check_same_n_vertices(self.maps_img.mesh, self.mask_img.mesh)
+            check_surface_data_ndims(self.mask_img, 1, "mask_img")
+
+        return self
+
+    def __sklearn_is_fitted__(self):
+        return hasattr(self, "n_elements_")
+
+    def _check_fitted(self):
+        if not self.__sklearn_is_fitted__():
+            raise ValueError(
+                f"It seems that {self.__class__.__name__} "
+                "has not been fitted."
+            )
+
+    @property
+    def _get_concatenated_maps_and_mask(self):
```

Comment:
@Remi-Gau, added a property to concatenate parts from mask and maps, used in both transform and inverse_transform methods.

Let me know if this is better.

Reply:
feels weird to have a property return a tuple

one private method and maybe one private property for each ?

Reply:
I feel like we do this across hemisphere concatenation fairly often, maybe a function in maskers/_utils would be better

Reply:
However, having a method within SurfaceImage like `.get_data()` would be the most ideal I think...
* relates to this comment: https://github.com/nilearn/nilearn/pull/4884#pullrequestreview-2482561037